src/containers/TriggerTemplate/TriggerTemplate.js

```diff
@@ -0,0 +1,377 @@
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import jsYaml from 'js-yaml';
+import './triggertemplate.scss';
+
+import {
+  Accordion,
+  AccordionItem,
+  Button,
+  DataTable,
+  InlineNotification
+} from 'carbon-components-react';
+import { injectIntl } from 'react-intl';
+import { Table as DashboardTable } from '@tektoncd/dashboard-components';
+import {
+  getSelectedNamespace,
+  getTriggerTemplate,
+  getTriggerTemplatesErrorMessage,
+  isWebSocketConnected
+} from '../../reducers';
+
+import { fetchTriggerTemplate } from '../../actions/triggerTemplates';
+
+const {
+  Table,
+  TableBody,
+  TableContainer,
+  TableHead,
+  TableHeader,
+  TableRow
+} = DataTable;
+
+const initialState = {
+  showYAML: false,
+  showDetails: false
+};
+
+export /* istanbul ignore next */ class TriggerTemplateContainer extends Component {
+  /* eslint-disable react/sort-comp */
+  constructor(props) {
+    super(props);
+    this.state = initialState;
+  }
+
+  // once redux store is available errors will be handled properly with dedicated components
+  static notification({ kind, message }) {
+    const titles = {
+      info: 'TriggerTemplate not available',
+      error: 'Error loading TriggerTemplate'
+    };
+    return (
+      <InlineNotification
+        kind={kind}
+        hideCloseButton
+        lowContrast
+        title={titles[kind]}
+        subtitle={message}
+      />
+    );
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { match, webSocketConnected } = this.props;
+    const { namespace, triggerTemplateName } = match.params;
+    const {
+      match: prevMatch,
+      webSocketConnected: prevWebSocketConnected
+    } = prevProps;
+    const {
+      namespace: prevNamespace,
+      triggerTemplateName: prevTriggerTemplateName
+    } = prevMatch.params;
+
+    if (
+      namespace !== prevNamespace ||
+      triggerTemplateName !== prevTriggerTemplateName ||
+      (webSocketConnected && prevWebSocketConnected === false)
+    ) {
+      this.fetchData();
+    }
+  }
+
+  /* istanbul ignore next */
+  handleYAML = () => {
+    this.setState(prevState => ({
+      showYAML: !prevState.showYAML,
+      showDetails: false
+    }));
+  };
+
+  /* istanbul ignore next */
+  handleDetails = () => {
+    this.setState(prevState => ({
+      showDetails: !prevState.showDetails,
+      showYAML: false
+    }));
+  };
+
+  fetchData() {
+    const { match } = this.props;
+    const { namespace, triggerTemplateName } = match.params;
+    this.props.fetchTriggerTemplate({ name: triggerTemplateName, namespace });
+  }
+
+  render() {
+    const {
+      intl,
+      error,
+      match,
+      selectedNamespace,
+      triggerTemplate
+    } = this.props;
+
+    if (error) {
+      return TriggerTemplateContainer.notification({
+        kind: 'error',
+        message: 'Error loading TriggerTemplate'
+      });
+    }
+
+    if (!triggerTemplate) {
```

Comment:
i feel it be better to show these error notifications in the previous page...

Reply:
Hmm, I think that'd be a bit weird though, because the end user would actually go to this TriggerTemplate page, fail (not sure why it ever would), and then navigate a user back and give the error 🤔